samples/algorithms/sudoku-grover/ColoringGroverWithConstraints.qs

```diff
@@ -113,257 +113,66 @@ namespace Microsoft.Quantum.Samples.ColoringGroverWithConstraints {
-        use startingColorConflictQubits = Qubit[nStartingColorConstraints];
-        use vertexColorConflictQubits = Qubit[numVertices];
-        within {
-            ConstrainByEdgeAndStartingColors(
-                colorsRegister, edges, startingColorConstraints,
-                edgeConflictQubits, startingColorConflictQubits, bitsPerColor
-            );
-            let zippedColorAndConfictQubit = Zipped(
-                Partitioned(ConstantArray(numVertices, bitsPerColor), colorsRegister),
-                vertexColorConflictQubits
-            );
-            for (color, conflictQubit) in zippedColorAndConfictQubit {
-                // Only allow colors from 0 to 8 i.e. if bit #3 = 1, then bits 2..0 must be 000.
-                use tempQubit = Qubit();
-                within {
-                    ApplyOrOracle(color[0 .. 2], tempQubit);
-                } apply{
-                    // AND color's most significant bit with OR of least significant bits.
-                    // This will set conflictQubit to 1 if color > 8.
-                    CCNOT(color[3], tempQubit, conflictQubit);
-                }
+            for ((start, end), conflictQubit) in Zipped(edges, edgeConflictQubits) {
+                // Check that endpoints of the edge have different colors:
+                // apply ColorEqualityOracle_Nbit oracle;
+                // if the colors are the same the result will be 1, indicating a conflict
+                ApplyColorEqualityOracle(
+                    colorsRegister[start * bitsPerColor .. (start + 1) * bitsPerColor - 1],
+                    colorsRegister[end * bitsPerColor .. (end + 1) * bitsPerColor - 1],
+                    conflictQubit
+                );
             }
         } apply {
             // If there are no conflicts (all qubits are in 0 state), the vertex coloring is valid.
-            ControlledOnInt(0, X)(edgeConflictQubits + startingColorConflictQubits + vertexColorConflictQubits, target);
+            ControlledOnInt(0, X)(edgeConflictQubits, target);
         }
     }
 
-    /// # Summary
-    /// OR oracle for an arbitrary number of qubits in query register.
-    ///
-    /// # Inputs
-    /// ## queryRegister
-    /// Qubit register to query.
-    /// ## target
-    /// Target qubit for storing oracle result.
-    operation ApplyOrOracle (queryRegister : Qubit[], target : Qubit) : Unit is Adj {
-        // x₀ ∨ x₁ = ¬ (¬x₀ ∧ ¬x₁)
-        // First, flip target if both qubits are in |0⟩ state.
-        ControlledOnInt(0, X)(queryRegister, target);
-        // Then flip target again to get negation.
-        X(target);
-    }
-
     /// # Summary
     /// Using Grover's search to find vertex coloring.
     ///
     /// # Input
-    /// ## numVertices
+    /// ## nVertices
     /// The number of Vertices in the graph.
     /// ## bitsPerColor
     /// The number of bits per color.
-    /// ## maxIterations
+    /// ## nIterations
     /// An estimate of the maximum iterations needed.
```

Comment:
Since we're updating the meaning of the parameter from the maximum number of iterations needed to the exact number done, let's also update the documentation